lib/gather/gather-single-telemetry.js

```diff
@@ -0,0 +1,45 @@
+const puppeteer = require('puppeteer');
+const { setTelemetry, getTelemetry } = require('../utils/telemetry');
+
+const DEFAULT_PUPPETEER_ARGS = { ignoreHTTPSErrors: true };
+
+module.exports = async function gatherSingleTelemetry(url, gatherFn, ...args) {
+  const browser = await puppeteer.launch(DEFAULT_PUPPETEER_ARGS);
+  const page = await browser.newPage();
+
+  await page.goto(url);
+
+  await page.exposeFunction('logErrorInNodeProcess', message => {
+    console.error(message); // eslint-disable-line no-console
+  });
+
+  const telemetry = await bridgeEvaluate(
+    page,
+    async (gFn, ...supportFns) => {
+      supportFns.forEach(fn => {
+        this[fn.name] = fn;
+      });
+      let telemetry = {};
+
+      telemetry = await gFn(...supportFns);
+      return telemetry;
+    },
+    gatherFn,
+    ...args
+  );
+
+  // get the current telemetry and add the `single-telemetry` key to it.
+  const currentTelemetry = getTelemetry();
+  currentTelemetry['single-telemetry'] = telemetry;
```

Comment:
I don't particularly like this nesting into the telemetry 🤔 

Maybe we can make this function take an options POJO instead, and have one of the options be to specify a custom telemetry name (passing `getTelemetry(name)` and `setTelemetry(name, value)`).

Reply:
done